test/spec/runners/shuffling.ex

```diff
@@ -0,0 +1,56 @@
+defmodule ShufflingTestRunner do
+  alias LambdaEthereumConsensus.StateTransition.Misc
+  use ExUnit.CaseTemplate
+  use TestRunner
+
+  @moduledoc """
+  Runner for Operations test cases. See: https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/shuffling
+  """
+
+  # Remove handler from here once you implement the corresponding functions
+  @disabled_handlers [
+    # "core"
+  ]
+
+  @impl TestRunner
+  def skip?(%SpecTestCase{} = testcase) do
+    Enum.member?(@disabled_handlers, testcase.handler)
+  end
+
+  @impl TestRunner
+  def run_test_case(%SpecTestCase{} = testcase) do
+    case_dir = SpecTestCase.dir(testcase)
+
+    %{seed: seed, count: index_count, mapping: indices} =
+      YamlElixir.read_from_file!(case_dir <> "/mapping.yaml")
+      |> SpecTestUtils.parse_yaml()
+
+    handle(testcase.handler, seed, index_count, indices)
+  end
+
+  defp handle("core", seed, index_count, indices) do
+    for index <- 0..(index_count - 1) do
+      result = Misc.compute_shuffled_index(index, index_count, seed)
+
+      case result do
+        {:ok, value} ->
+          assert value == Enum.fetch!(indices, index)
+
+        {:error, _} ->
+          assert index >= index_count or index_count == 0
+      end
+    end
+
+    for index <- (index_count - 1)..0 do
+      result = Misc.compute_shuffled_index(index, index_count, seed)
+
+      case result do
+        {:ok, value} ->
+          assert value == Enum.fetch!(indices, index)
+
+        {:error, _} ->
+          assert index >= index_count or index_count == 0
+      end
+    end
```

Comment:
What's the difference between these two cases? Wouldn't they be always equivalent since the function is pure?

Reply:
It's specified in the spec that should be checked and it kinda shows that function is independent of the order.

Reply:
Could you share a link to that? Also, we should add a code comment with this info.

Reply:
https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/shuffling. Number 2

Reply:
We should add a comment with that info, for example:

> NOTE: this is the same as before, but the test-format specs suggest including this check